Detectors/TPC/workflow/src/CalibdEdxSpec.cxx

```diff
@@ -0,0 +1,132 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "TPCWorkflow/CalibdEdxSpec.h"
+
+#include <vector>
+#include <memory>
+
+//o2 includes
+#include "CCDB/CcdbApi.h"
+#include "CCDB/CcdbObjectInfo.h"
+#include "DataFormatsTPC/TrackTPC.h"
+#include "DetectorsCalibration/Utils.h"
+#include "Framework/Task.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "TPCCalibration/CalibdEdx.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace tpc
+{
+
+class CalibdEdxDevice : public Task
+{
+ public:
+  void init(framework::InitContext& ic) final
+  {
+    int slot_length = ic.options().get<int>("tf-per-slot");
+    int max_delay = ic.options().get<int>("max-delay");
+    int min_entries = std::max(50, ic.options().get<int>("min-entries"));
+    double min_p = std::min(0.49, ic.options().get<double>("min-momentum"));
+    double max_p = std::max(0.51, ic.options().get<double>("max-momentum"));
+    int min_clusters = std::max(15, ic.options().get<int>("min-clusters"));
+    int nbins = std::max(50, ic.options().get<int>("nbins"));
+
+    mCalibrator = std::make_unique<tpc::CalibratordEdx>(min_entries, min_p, max_p, min_clusters, nbins);
+    mCalibrator->setSlotLength(slot_length);
+    mCalibrator->setMaxSlotsDelay(max_delay);
+  }
+
+  void run(ProcessingContext& pc) final
+  {
+    auto tfcounter = o2::header::get<DataProcessingHeader*>(pc.inputs().get("tracks").header)->startTime;
+    auto tracks = pc.inputs().get<gsl::span<tpc::TrackTPC>>("tracks");
+
+    LOG(INFO) << "Processing TF " << tfcounter << " with " << tracks.size() << " tracks";
+
+    mCalibrator->process(tfcounter, tracks);
+    sendOutput(pc.outputs());
+
+    const auto& infoVec = mCalibrator->getInfoVector();
+    LOG(INFO) << "Created " << infoVec.size() << " objects for TF " << tfcounter;
+  }
+
+  void endOfStream(EndOfStreamContext& eos) final
+  {
+    LOG(INFO) << "Finalizing calibration";
+    // FIXME: not sure about this
+    constexpr uint64_t INFINITE_TF = 0xffffffffffffffff;
+    mCalibrator->checkSlotsToFinalize(INFINITE_TF);
+    sendOutput(eos.outputs());
+    mCalibrator->dumpToFile("mip_position.root");
```

Comment:
Please add an option that this is only dumped if activated in the workflow, like here https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/TPC/workflow/include/TPCWorkflow/TPCCalibPedestalSpec.h#L180